### ISSUE TITLE
Question about VQA benchmark reproducing

### ISSUE THREAD

**Issue by haibara-ai** [Go2Heart/StreamFormer#9]
Hi, I'm reproducing the StreamFormer metric in VideoQA tasks, e.g. `VideoMME`. When I finished all 3 stages finetuning, the evaluation result of VideoMME only got score about 9.778%. I checked the eval log in lmms_eval, for many questions, the answer is empty. e.g.
```
2025-09-24 22:57:29.090 | DEBUG    | lmms_eval.models.llava_vid:generate_until:525 - Question: <image>
Select the best answer to the following multiple-choice question based on the video and the subtitles. Respond with only the letter (A, B, C, or D) of the correct option.
Why does the male protagonist, wearing a white shirt and black shorts in the video, take painkillers during the race?
A. He suffered a stress fracture in his foot during the competition.
B. Because his knee was suddenly injured during the competition.
C. Because he injured his foot during previous training, running during the competition would be very painful.
D. Because he suffered a stress fracture in his foot during previous training.
The best answer is:
2025-09-24 22:57:29.090 | DEBUG    | lmms_eval.models.llava_vid:generate_until:526 - Answer:
```
However when I checked the stage2 checkpoint, theVideoMME score could get score about 35%. Do you have any suggestion on this problem? Thanks.
Here is the stage3  training command:
```
export OMP_NUM_THREADS=8
export NCCL_IB_DISABLE=0
export NCCL_IB_GID_INDEX=3
export NCCL_SOCKET_IFNAME=eth0
export RANK=0
export NUM_GPUS=8
export NNODES=1
export ADDR=localhost
export NCCL_DEBUG=INFO
export PORT=12324

LLM_VERSION="checkpoints/vicuna-7b-v1.5"
LLM_VERSION_CLEAN="${LLM_VERSION//\//_}"
VISION_MODEL_VERSION="checkpoints/streamformer-timesformer"
VISION_MODEL_VERSION_CLEAN="${VISION_MODEL_VERSION//\//_}"

############### Pretrain ################

PROMPT_VERSION="v1"

BASE_RUN_NAME="llavanext-${VISION_MODEL_VERSION_CLEAN}-${LLM_VERSION_CLEAN}-short-010-v2"
echo "BASE_RUN_NAME: ${BASE_RUN_NAME}"
MID_RUN_NAME="MID_${BASE_RUN_NAME}"

ACCELERATE_CPU_AFFINITY=1 torchrun --nproc_per_node="${NUM_GPUS}" --nnodes="${NNODES}" --node_rank="${RANK}" --master_addr="${ADDR}" --master_port="${PORT}" \
    llava/train/train_mem.py \
    --deepspeed scripts/zero2.json \
    --model_name_or_path checkpoints/MID_llavanext-checkpoints_streamformer-timesformer-checkpoints_vicuna-7b-v1.5-mlp2x_gelu-pretrain_blip558k_plain_frozen_v-frame-16_all \
    --version ${PROMPT_VERSION} \
    --data_path data/llava-video-178k/selected_videos.yaml\
    --image_folder data/llava-video-178k/ \
    --video_folder data/llava-video-178k/ \
    --mm_tunable_parts="mm_mlp_adapter,mm_language_model" \
    --mm_vision_tower_lr=2e-6 \
    --vision_tower ${VISION_MODEL_VERSION} \
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --group_by_modality_length True \
    --image_aspect_ratio anyres \
    --image_grid_pinpoints "[(224, 448), (448, 224), (448, 448), (672, 224), (224, 672)]" \
    --mm_patch_merge_type spatial_unpad \
    --bf16 True \
    --run_name $MID_RUN_NAME \
    --output_dir "checkpoints/${MID_RUN_NAME}" \
    --num_train_epochs 1 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 4 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 500 \
    --save_total_limit 1 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 4096 \
    --gradient_checkpointing True \
    --dataloader_num_workers 2 \
    --lazy_preprocess True \
    --report_to tensorboard \
    --torch_compile True \
    --torch_compile_backend "inductor" \
    --dataloader_drop_last True \
    --attn_implementation flash_attention_2 \
    --mm_newline_position "no_token" \
    --mm_spatial_pool_stride 2 \
    --frames_upbound 16 \
    --force_sample True
```

and the training log of stage3.
[trainer_state.json](https://github.com/user-attachments/files/22529441/trainer_state.json)

**Comment by Go2Heart** [comment]
That is weired. I have uploaded my [trainer_state.json](https://github.com/user-attachments/files/22529865/trainer_state.json), and you can also have a look. I believe the training script and log seem correct. What version of the code are you using?

**Comment by Go2Heart** [comment]
There is one recent modification in our code to make the video QA model streaming and support for KV cache. What's your commit id for your training code?

**Comment by Go2Heart** [comment]
> During the stage3 training, there are lots of log like
> 
> ```
> [h264 @ 0x59a23f000c00] mmco: unref short failure
> [h264 @ 0x59a23f000c00] mmco: unref short failure
> [h264 @ 0x59a23f000c00] mmco: unref short failure
> [h264 @ 0x59a23f000c00] mmco: unref short failure
> [h264 @ 0x59a23a743080] mmco: unref short failure
> [h264 @ 0x59a23a743080] mmco: unref short failure
> ```
> 
> Is this expected?

This is expected for decord loading some videos, see [here](https://github.com/dmlc/decord/issues/177)

**Comment by haibara-ai** [comment]
Tried the [8780965](https://github.com/Go2Heart/StreamFormer/commit/878096585132679d017104c407bcb991d824449b) and [4686176](https://github.com/Go2Heart/StreamFormer/commit/4686176035bf2f198972ccef31ec69084334899c) several times, it looks like the different commit didn't make too much difference. But the results are instable cross different runs, the best results on videomme is about 43% and mlvu_dev is about 48%, while the worst one could go down to about 20~30% on videomme and mlvu_dev. Did you ever see such issue?

**Comment by haibara-ai** [comment]
Thanks for your reply.
